core/io/http_client_tcp.cpp

```diff
@@ -438,8 +438,7 @@ Error HTTPClientTCP::poll() {
 			return OK;
 		} break;
 		case STATUS_REQUESTING: {
-			if (request_buffer->get_available_bytes()) {
-				int avail = request_buffer->get_available_bytes();
+			if (size_t avail = request_buffer->get_available_bytes()) {
```

Comment:
What are you doing here? Declaring a variable inside of an if statement?

Reply:
Sorry for stepping in. That's a thing you can do since C++14 (?). As much as I like it, to help scoping variables, the current coding style forbids it.

Reply:
Can you point me to where it says its forbidden because I didnt find it ?

Reply:
I can change the coding guideline to be explicit about it. Its very sparse right now

Reply:
Updating the style would be a major work handled by maintainers, so not something for a first contributor, but thank you

It also requires some discussion between teams